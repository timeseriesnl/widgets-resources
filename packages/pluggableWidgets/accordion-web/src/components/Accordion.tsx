import { createElement, ReactElement, useReducer, useRef } from "react";

import AccordionGroup, { AccGroup } from "./AccordionGroup";
import { getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
import { AccordionContainerProps } from "../../typings/AccordionProps";
import classNames from "classnames";

export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
    id: string;
    groups: AccGroup[];
    collapsible: boolean;
    singleExpandedGroup?: boolean;
}

export default function Accordion(props: AccordionProps): ReactElement | null {
    const previousGroupsPropValue = useRef(props.groups);

    const [accordionGroups, accordionGroupsDispatch] = useReducer(
        getAccordionGroupsReducer(props.collapsible ? (props.singleExpandedGroup ? "single" : "multiple") : "all"), // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
        props.groups.map(group => ({ ...group, collapsed: props.collapsible }))
    );

    if (props.groups !== previousGroupsPropValue.current) {
        previousGroupsPropValue.current = props.groups;
        accordionGroupsDispatch({ type: "sync", groups: props.groups });
    }

    const accordionGroupElements = accordionGroups.map((group, index) => (
        <AccordionGroup
            key={index}
            group={group}
            accordionGroupsDispatch={props.collapsible ? accordionGroupsDispatch : undefined}
        />
    ));

    return (
        <div
            id={props.id}
            className={classNames("widget-accordion", props.class)}
            style={props.style}
            tabIndex={props.tabIndex}
        >
            {accordionGroupElements}
        </div>
    );
}
