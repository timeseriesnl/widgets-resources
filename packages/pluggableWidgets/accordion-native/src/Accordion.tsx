import { createElement, ReactElement, useState, useRef } from "react";
import { View } from "react-native";
import { flattenStyles } from "@mendix/piw-native-utils-internal";
import { executeAction, isAvailable } from "@mendix/piw-utils-internal";

import { AccordionProps, GroupsType } from "../typings/AccordionProps";
import { defaultAccordionStyle, AccordionStyle } from "./ui/Styles";
import { AccordionGroup } from "./components/AccordionGroup";

export type Props = AccordionProps<AccordionStyle>;

export function Accordion(props: Props): ReactElement | null {
    const styles = flattenStyles(defaultAccordionStyle, props.style);
    const initialRender = useRef(false);
    const [expandedGroups, setExpandedGroups] = useState<number[]>([]);

    if (!initialRender.current) {
        let initialExpandedGroups = props.groups.reduce(
            (acc: number[], group: GroupsType, index: number): number[] =>
                group.groupCollapsed === "groupStartExpanded" ? [...acc, index] : acc,
            []
        );
        initialExpandedGroups = !props.collapsible
            ? initialExpandedGroups
            : props.collapseBehavior === "singleExpanded"
            ? [initialExpandedGroups[0]]
            : initialExpandedGroups;
        setExpandedGroups(initialExpandedGroups);
        initialRender.current = true;
    }

    const onPressGroupHeader = (group: GroupsType, index: number): void => {
        const expanded = expandedGroups.includes(index);
        if (group.groupAttribute && isAvailable(group.groupAttribute)) {
            group.groupAttribute?.setValue(!expanded);
        }
        executeAction(group.groupOnChange);
        expanded ? collapseGroup(index) : expandGroup(index);
    };

    const collapseGroup = (index: number): void => {
        setExpandedGroups(expandedGroups.filter(i => i !== index));
    };
    const expandGroup = (index: number): void => {
        setExpandedGroups(props.collapseBehavior === "singleExpanded" ? [index] : [...expandedGroups, index]);
    };

    return (
        <View style={styles.container}>
            {props.groups.map(
                (group: GroupsType, index: number): ReactElement => (
                    <AccordionGroup
                        key={index}
                        index={index}
                        collapsible={props.collapsible}
                        icon={props.icon}
                        iconCollapsed={props.iconCollapsed}
                        iconExpanded={props.iconExpanded}
                        group={group}
                        isExpanded={expandedGroups.includes(index)}
                        expandGroup={expandGroup}
                        onPressGroupHeader={onPressGroupHeader}
                        visible={group.visible}
                        style={styles.group}
                    />
                )
            )}
        </View>
    );
}
