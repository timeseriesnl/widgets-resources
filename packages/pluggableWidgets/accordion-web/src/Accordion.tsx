import { createElement, ReactElement, useMemo } from "react";
import { ValueStatus } from "mendix";

import { Accordion as AccordionComponent, AccordionGroups } from "./components/Accordion";
import { Header } from "./components/Header";
import { useIconGenerator } from "./utils/iconGenerator";

import { AccordionContainerProps, GroupsType } from "../typings/AccordionProps";

export function Accordion(props: AccordionContainerProps): ReactElement | null {
    const groups: AccordionGroups | undefined = useMemo(() => translateGroups(props.groups), [props.groups]);

    const generateIcon = useIconGenerator(
        props.advancedMode,
        props.animateIcon,
        { data: props.icon?.value, loading: props.icon?.status === ValueStatus.Loading },
        { data: props.expandIcon?.value, loading: props.expandIcon?.status === ValueStatus.Loading },
        { data: props.collapseIcon?.value, loading: props.collapseIcon?.status === ValueStatus.Loading }
    );

    if (!groups) {
        return null;
    }

    return (
        <AccordionComponent
            id={props.name}
            class={props.class}
            style={props.style}
            tabIndex={props.tabIndex}
            groups={groups}
            collapsible={props.collapsible}
            animateContent={props.animate}
            singleExpandedGroup={props.collapsible ? props.expandBehavior === "singleExpanded" : undefined}
            generateHeaderIcon={generateIcon}
            showGroupHeaderIcon={props.showIcon}
        />
    );
}

function translateGroups(groups: AccordionContainerProps["groups"]): AccordionGroups | undefined {
    if (someGroupMissingData(groups)) {
        return undefined;
    }

    return groups.map(group => {
        let header = group.headerContent;

        if (group.headerRenderMode === "text") {
            header = <Header heading={group.headerHeading}>{group.headerText.value}</Header>;
        }

        return {
            header,
            content: group.content,
            collapsed: group.collapsed?.value,
            initiallyCollapsed:
                group.initialCollapsedState === "dynamic"
                    ? group.initiallyCollapsed.value
                    : group.initialCollapsedState === "collapsed",
            visible: group.visible.value!,
            dynamicClassName: group.dynamicClass?.value,
            onToggleCompletion: group.collapsed?.setValue
        };
    });
}

function someGroupMissingData(groups: GroupsType[]): boolean {
    return groups.some(
        group =>
            group.visible.value === undefined ||
            group.headerText.value === undefined ||
            group.initiallyCollapsed.value === undefined ||
            (group.collapsed && group.collapsed.value === undefined)
    );
}
