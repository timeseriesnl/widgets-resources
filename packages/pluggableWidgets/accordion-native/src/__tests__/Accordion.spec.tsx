import { actionValue, dynamicValue, EditableValueBuilder } from "@mendix/piw-utils-internal";
import { createElement } from "react";
import { Text } from "react-native";
import { Accordion, Props } from "../Accordion";
import { fireEvent, render } from "react-native-testing-library";

// jest.mock("react-native/Libraries/Utilities/Platform", () => ({
//     OS: "ios",
//     select: jest.fn(dict => dict.ios),
// }));

const defaultProps: Props = {
    name: "accordion-test",
    style: [],
    groups: [
        {
            headerRenderMode: "text",
            headerTextRenderMode: "heading1",
            headerText: dynamicValue<string>("Header1"),
            headerContent: undefined,
            content: <Text>Content</Text>,
            visible: dynamicValue<boolean>(true),
            groupCollapsed: "groupStartExpanded",
            groupCollapsedDynamic: dynamicValue<boolean>(undefined),
            groupAttribute: new EditableValueBuilder<boolean>().withValue(undefined).build(),
            groupOnChange: actionValue()
        },
        {
            headerRenderMode: "custom",
            headerTextRenderMode: "heading1",
            headerText: dynamicValue<string>(undefined),
            headerContent: <Text>Header2</Text>,
            content: <Text>Content</Text>,
            visible: dynamicValue<boolean>(true),
            groupCollapsed: "groupStartExpanded",
            groupCollapsedDynamic: dynamicValue<boolean>(undefined),
            groupAttribute: new EditableValueBuilder<boolean>().withValue(undefined).build(),
            groupOnChange: actionValue()
        }
    ],
    collapsible: true,
    collapseBehavior: "singleExpanded",
    icon: "right",
    iconExpanded: undefined,
    iconCollapsed: undefined
};

describe("Accordion", () => {
    describe("in collapsible & single expanded group mode", () => {
        it("renders correctly", () => {
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("renders the accordion groups icon on left side", () => {
            const accordion = render(<Accordion {...defaultProps} icon={"left"} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("renders the accordion groups without an icon", () => {
            const accordion = render(<Accordion {...defaultProps} icon={"no"} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("renders the first accordion group initially collapsed", () => {
            defaultProps.groups[0].groupCollapsed = "groupStartCollapsed";
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("hides the first accordion group when visibility is false", () => {
            defaultProps.groups[0].visible = dynamicValue(false);
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("renders the first accordion group initially collapsed based on expression", () => {
            defaultProps.groups[0].groupCollapsed = "groupStartDynamic";
            defaultProps.groups[0].groupCollapsedDynamic = dynamicValue(true);
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("renders the first accordion group initially collapsed based on attribute", () => {
            defaultProps.groups[0].groupAttribute = new EditableValueBuilder<boolean>().withValue(true).build();
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });

        it("executes the first accordion group onChange action when header pressed", () => {
            const accordion = render(<Accordion {...defaultProps} />);
            fireEvent.press(accordion.getByText("Header1"));
            expect(defaultProps.groups[0].groupOnChange?.execute).toHaveBeenCalledTimes(1);
        });
    });

    describe("in collapsible & multi expanded group mode", () => {
        it("renders correctly", () => {
            defaultProps.collapseBehavior = "multipleExpanded";
            const accordion = render(<Accordion {...defaultProps} />);
            expect(accordion.toJSON()).toMatchSnapshot();
        });
    });
});
