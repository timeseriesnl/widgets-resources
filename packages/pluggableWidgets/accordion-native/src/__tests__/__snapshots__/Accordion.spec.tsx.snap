// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion in collapsible & multi expanded group mode renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode hides the first accordion group when visibility is false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders the accordion groups icon on left side 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          Object {
            "flexDirection": "row-reverse",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          Object {
            "flexDirection": "row-reverse",
          },
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders the accordion groups without an icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders the first accordion group initially collapsed 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders the first accordion group initially collapsed based on attribute 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Accordion in collapsible & single expanded group mode renders the first accordion group initially collapsed based on expression 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 40,
            "fontWeight": "600",
            "lineHeight": 60,
          }
        }
      >
        Header1
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      <Text>
        Header2
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          testId="icon"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
            "paddingTop": 8,
          }
        }
      >
        <Text>
          Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;
